registry/storage/driver/s3-aws/s3.go

```diff
@@ -695,7 +695,19 @@ func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) e
 	if err := d.copy(ctx, sourcePath, destPath); err != nil {
 		return err
 	}
-	return d.Delete(ctx, sourcePath)
+	attempt := 30
+	attemptRelay := 1000 // relay 500ms before next attempt
```

Comment:
We can add a unity func for retry as other driver may have the same requirement, and the duration should be a dynamic, it's better to get it from the back off func, like:

```
timeout := time.After(opts.Timeout)
for attempt := 1; ; attempt++ {
    select {
    case <-timeout:
	    return ErrRetryTimeout
    default:
	    if err := f(); err != nil {
		    sleep := getBackoff(attempt, opts.InitialInterval, opts.MaxInterval, true)
		    if opts.Callback != nil {
			    opts.Callback(err, sleep)
		    }
    
		    time.Sleep(sleep)
	    } else {
		    return nil
	    }
    }
}
```
And for the backoff, there is a article can have a reference. https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/

Reply:
Really appreciate your advice! I'm new to Go, I'll look into it!